### PR TITLE
Render a notice when the available balance is below the minimum deposit threshold

### LINKED ISSUE #6774: Show clear info in UI when there will be no payout because merchant balance is below deposit minimums (e.g. AUD$5)
### Description

The estimated deposit date displayed for users in WooCommerce Payments may draw confusion when the minimum deposit amount is not yet met. 

https://d.pr/i/r3y5wm+
For example: The minimum deposit amount for AUD currency is $5, as stated in the [doceumentation](https://woocommerce.com/document/woocommerce-payments/deposits/deposit-schedule/#minimum-deposit-amounts). However, it is worth noting that the dashboard does not provide a clear indication of this requirement. As a result, users may be left wondering why their deposit did not go through on the estimated date.

To improve the user experience, we can incorporate a tooltip or a similar prompt when the deposit amount is below the minimum requirement. This would explain that the deposit will be processed once the minimum balance is exceeded. Additionally, we can include a note about minimum balances in a prominent location in the user interface

### Additional context

- [Slack conversation](https://a8c.slack.com/archives/C7U3Y3VMY/p1689309505749639)
- 6535410-zen

**Update**
A proposed solution for this issue is a banner on the **Payments → Overview** screen, as per @bradyv's [design](https://www.figma.com/file/aDFNTeogq2cXSr4Nqeni8g/%5BWCPay%5D-Instant-Deposits?type=design&node-id=294-83778&mode=design&t=exSeHwn5tRxr33zg-4).

When the available balance is below the minimum deposit threshold:
* Remove the schedule string as it causes confusion when deposits aren't happening
* Show the banner below
      image

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #6774

### Changes proposed in this Pull Request

This PR renders a notice on the **Payments → Overview → Deposits** card when the available balance for the selected currency is above zero yet below the minimum deposit amount.

Minimum deposit amounts are defined per currency and made available to the UI via the `wcpaySettings` global JS object.

**TODO:**
- [x] Testing instructions
- [x] Check on mobile
- [x] Check against server PR https://github.com/Automattic/woocommerce-payments-server/pull/4308

### Testing instructions

* Test the following scenarios and visit the Payments → Overview screen to ensure the notice appears when expected:

**Available balance below minimum deposit threshold**
* A notice should appear with the correct minimum balance shown for the selected currency.
* The deposit schedule message should not appear
image

**Available balance above minimum deposit threshold**
* The minimum deposit threshold notice should **not** appear.
* The deposit schedule message should appear.
image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : **Not provided due to low risk and lengthy setup**
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable. **N/A**
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable. **Added to what's new post**

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7710` or branch name `fix/6774-deposit-minimum-threshold-ui` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/6774-deposit-minimum-threshold-ui) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: c94613d4b6daa10904a1ae451bdf926e54909d7e
- Build time: 2024-02-05 22:31:54 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/inline-notice/styles.scss [hunk 1]
```diff
@@ -31,6 +31,11 @@
 	.components-notice__content {
 		margin-top: 2px;
 		margin-bottom: 2px;
+
+		// Ensure links don't wrap across lines, e.g. "Learn (newline) more".
+		a {
+			white-space: nowrap;
+		}
```
Comment (Jinksi) [inline client/components/inline-notice/styles.scss:38]: I think this is a general enhancement for this component.

Example, before, the link element within the inline notice could wrap across multiple lines:
image

After: 
image

**Review by brucealdridge — APPROVED** [review]
Reviewed code and tested. Working fine. Noticed documentation is incorrect so will follow up outside this PR.

**Comment by brucealdridge** [comment]
While the code is `ready to merge` there is an issue with the underlying data.

The minimum deposit amounts used are WooPayments limits for manual deposits. Stripe has separate limits for scheduled/automated deposits.

I'll look into how we can resolve this problem and whether we can merge this code as is and make the relevant changes on the server or if there are other concerns.

**Comment by haszari** [comment]
> I'll look into how we can resolve this problem and whether we can merge this code as is and make the relevant changes on the server or if there are other concerns.

Sounds like a blocker – good spotting @brucealdridge ! Added blocker label until resolved, i.e. we don't want to merge this accidentally. 

We might be able to tweak this PR to focus on the negative balance case, and then look at minimums for scheduled deposits as a follow up. Reading through the discussion, I'm not clear if scheduled deposits are affected by minimums or not – might have been an incorrect assumption in the original bug report?

**Comment by Jinksi** [comment]
> We might be able to tweak this PR to focus on the negative balance case

We currently have a notice appear when the available balance is negative (https://github.com/Automattic/woocommerce-payments/pull/7656#:~:text=Negative%20available%20balance%20%E2%80%93%20Deposit%20UI%20rendered%20with%20notice). (I assume this is what you're referring to).

image

**Comment by Jinksi** [comment]
The automated review request for @Automattic/harmony can be safely ignored, although I can't cancel the review request.

Something went awry when I was fixing merge conflicts I guess 🤷‍♂️ 

Screenshot 2024-01-30 at 16 27 43

**Pull request merged**